### PR TITLE
Adds shared comments feature

### PR DESCRIPTION
**Pull request by victorhuangwq**
## Description

Adds ability for instructors and CAs to have shared comments. Graders can easily opt in to have their comments be used by other graders.

## Motivation and Context

From interview with 122 TAs.

## How Has This Been Tested?

Pool here will refers to the list of shared comments shown

- Check add to pool, and save. Click edit to see that it is unchecked.
- Remove add to pool and save, able to see that being updated (unchecked, and no longer in the pool in creating new annotations)
- Delete annotation. See that it is no longer in the pool
- Edit annotation. See that it is updated in the pool
- Refresh to see if the change persists

Other cases:
- Act as a course assistant, and I'm still able to see the shared comments
- Change to a different assessment, I should not be able to see the shared comment from other assessments

Note: As of now, it does not appear when you are editing an annotations. To be discussed if that's what we want.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

- [ ] Question: Do you think we should have a tool tip to tell them what the comment pool means?

### CODE REVIEW

**Comment by michellexliu** [comment]
The major functionality all seems to be working well! There is a small UI bug where the comment pool doesn't stick to the input box when you click into it and then scroll:
Screen Shot 2022-04-18 at 11 04 56 PM

Also, this isn't a major issue & is something that already wasn't working before this PR, but right now the annotations don't preserve line breaks (low priority issue if we want to ship this out fast).

**Comment by victorhuangwq** [comment]
I definitely didn't notice that the line breaks wasn't preserved. Let me see if I can fix it.

**Comment by victorhuangwq** [comment]
- Adds tooltip
image
- Adds autocomplete to editing
- Allow newlines in annotations

File: app/assets/stylesheets/annotations.css [hunk 1]
```diff
@@ -794,3 +812,53 @@
...
+  max-height: 15rem;
+  max-width: 60%;
+  overflow-y: auto;
+  /* prevent horizontal scrollbar */
+  overflow-x: hidden;
+}
+
+/* Golden Layout */
+
+.lm_content {
```
Comment (victorhuangwq) [inline app/assets/stylesheets/annotations.css:839]: Here

**Review by michellexliu — APPROVED** [review]
lgtm!

**Pull request merged**